src/linear_algebra/affine_space/affine_subspace.lean

```diff
@@ -0,0 +1,1083 @@
+/-
+Copyright (c) 2020 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Joseph Myers.
+-/
+import linear_algebra.affine_space.basic
+import linear_algebra.tensor_product
+import data.set.intervals.unordered_interval
+
+/-!
+# Affine spaces
+
+This file defines affine subspaces (over modules) and the affine span of a set of
+points.  For affine combinations of points, see `linear_algebra.affine_space.combination`.  For
+affinely independent families of points, see `linear_algebra.affine_space.independent`.  For some
+additional results relating to finite-dimensional subspaces of affine spaces, see
+`linear_algebra.affine_space.finite_dimensional`.
```

Comment:
I don't think this summary of what's in what file belongs in `linear_algebra.affine_space.affine_subspace`. Now that we have `linear_algebra.affine_space.basic` again, it could go there (with additional pointers to the `affine_subspace` and `affine_map` files, and others that have been added without being mentioned there), if we wish to keep such an overall summary of the various affine space pieces and where they are.

Otherwise I'm fine with how the documentation is now arranged.